api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -321,6 +321,10 @@ const (
 )
 
 // HostedClusterSpec is the desired behavior of a HostedCluster.
+// +kubebuilder:validation:XValidation:rule="self.platform.type == 'Azure' ? self.services.exists(s, s.service == 'APIServer' && s.servicePublishingStrategy.type == 'Route' && s.servicePublishingStrategy.route.hostname != '') : true",message="Azure platform requires APIServer Route service with a hostname to be defined"
```

Comment:
I think this is true for all platforms, not only Azure. If you're using a route for APIserver you need to specify the hostname

Reply:
ack, once we get this in with agreement on the proposed testing pattern I'll follow up for increase/improve validations.